poutyne/framework/metrics/epoch_metrics.py

```diff
@@ -0,0 +1,139 @@
+        If the arguments is of type integer, The score for this class (the label number) is calculated. Otherwise, this
+        determines the type of averaging performed on all the data:
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted mean.
+            This does not take label imbalance into account.
+
+    Attributes:
+        Beta (float): The averaging of recall versus precision.
+        Average (str) or (int): The method to calculate the F-score.
+
+    """
+
+    def __init__(self, beta=1.0, average='micro'):
+        super().__init__()
+        if sys.version_info[1] < 3.6:
+            raise NotImplementedError("allen nlp don't support python version older than 3.6.1.")
+        if FBetaMeasure is None:
+            raise ImportError("allen nlp need to be installed to use this class.")
+
+        self.label = None
+        if isinstance(average, int):
+            self.label = average
+            self.running_measure = FBetaMeasure.FBetaMeasure(beta=beta)
+        else:
+            self.running_measure = FBetaMeasure.FBetaMeasure(beta=beta, average=average)
+
+    def __call__(self, y_prediction, y_true):
+        """
+        Update the confusion matrix for calculating the F-score.
+
+        Args:
+            y_prediction : Prediction of the model.
+            y_true : A tensor of the gold labels. Can also be a tuple of gold_label and a mask.
+        """
+
+        mask = None
+        if isinstance(y_true, tuple):
+            y_true, mask = y_true
+        self.running_measure(y_prediction, y_true, mask=mask)
+
+    def get_metric(self, reset=True):
+        """
+        Method to get the metric score.
+
+        Args:
+            reset (Bool) : To either or not reset the confusion matrix after calculating the F-score. Default value
+            is True.
+        """
+
+        if self.label is not None:
+            return self.running_measure.get_metric(reset=reset)['fscore'][self.label]
+        return self.running_measure.get_metric(reset=reset)['fscore']
+
+
+class F1(FBeta):
+    """
+    Wrapper around the Allen NLP FBetaMeasure class in the specific case where beta is equal to one (1).
+
+    Args:
+        average : (str) or (int), ['micro' (default), 'macro', label_number]
+        If the arguments is of type integer, The score for this class (the label number) is calculated. Otherwise, this
+        determines the type of averaging performed on all the data:
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted mean.
+            This does not take label imbalance into account.
+
+    Attributes:
```

Comment:
Same as above, Average isn't an attribute of the class.

Reply:
True also for the beta. Mistake.